src/PolygonDraw/PolygonVertex.tsx

```diff
@@ -21,93 +19,99 @@ export interface Props {
     onDragEnd: () => void;
 }
 
-export interface State {
-    isHovered: boolean;
-    isDragged: boolean;
-    latLng: LatLng;
-}
-
-export class PolygonVertex extends React.Component<Props, State> {
-    private circleMarkerElement: CircleMarker;
-    state = {
-        isHovered: false,
-        isDragged: false,
-        latLng: new LatLng(0, 0),
-    };
+export const PolygonVertex: FC<Props> = ({
+    coordinate,
+    isSelected,
+    index,
+    onClick,
+    onDrag,
+    onDragEnd,
+    onDragStart,
+}) => {
+    const [isHovered, setIsHovered] = useState(false);
+    const [isDragged, setIsDragged] = useState(false);
+    const [latLng, setLatLng] = useState(new LatLng(0, 0));
 
-    static getDerivedStateFromProps = (props: Props, state: State) => {
-        return state.isDragged ? state : { ...state, latLng: createLeafletLatLngFromCoordinate(props.coordinate) };
-    };
+    let circleMarkerElement: CircleMarker;
 
-    componentDidMount() {
-        if (this.circleMarkerElement) {
-            this.circleMarkerElement.on('dragstart', this.handleDragStart);
-            this.circleMarkerElement.on('drag', this.handleDrag);
-            this.circleMarkerElement.on('dragend', this.handleDragEnd);
+    useEffect(() => {
+        if (!isDragged && !!coordinate) {
+            setLatLng(createLeafletLatLngFromCoordinate(coordinate));
         }
-    }
+    }, [isDragged, coordinate]);
 
-    componentWillUnmount() {
-        if (this.circleMarkerElement) {
-            this.circleMarkerElement.off('dragstart', this.handleDragStart);
-            this.circleMarkerElement.off('drag', this.handleDrag);
-            this.circleMarkerElement.off('dragend', this.handleDragEnd);
-        }
-    }
+    const handleDragStart = useCallback(
+        (event: DragEvent) => {
+            onDragStart(event.target.getLatLng(), index);
+            setIsDragged(true);
+        },
+        [index, onDragStart]
+    );
 
-    // eslint-disable-next-line
-    setCircleMarkerRef = (ref: any) => {
-        if (ref) {
-            this.circleMarkerElement = ref;
-        }
-    };
+    const handleDrag = useCallback(
+        (event: DragEvent) => {
+            if (isDragged) {
+                onDrag(event.target.getLatLng());
+            }
+        },
+        [isDragged, onDrag]
+    );
 
-    handleClick = () => this.props.onClick(this.props.index);
-    handleMouseOver = () => this.setState({ isHovered: true });
-    handleMouseOut = () => this.setState({ isHovered: false });
+    const handleDragEnd = useCallback(() => {
+        onDragEnd();
+        setIsDragged(false);
+    }, [onDragEnd]);
 
-    handleDragStart = (event: DragEvent) => {
-        this.props.onDragStart(event.target.getLatLng(), this.props.index);
-        this.setState({ isDragged: true });
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const setCircleMarkerRef = (ref: any) => {
```

Comment:
Could you help me to understand what is going on in this method? I see we have `cicrleMarkerElement` variable which is created on every render, and we assign something unknown which is called `ref`. Why are we doing this?

Reply:
Yes this one seems outdated and can cause bugs - I will give it another try using useRef and a different setter approach.

Reply:
I added a fix in this commit: #[fcf9656](https://github.com/freenowtech/react-polygon-editor/pull/27/commits/fcf96560e8d151c524eaf47654efc80e2a7f9a33)